File: algo/50. Pow(x, n).js
/**
 * @param {number} x
 * @param {number} n
 * @return {number}
 */
var myPow = function(x, n) {
    if (n==0) {
        return 1;
    }
    if (n<0) {
        n = -n;
        x = 1/x;
    }
    return n%2==0 ? myPow(x*x, n/2) : x*myPow(x*x, (n-1)/2);
};

File: round3/50. Pow(x, n).cs
public class Solution {
    public double MyPow(double x, int n) {
        if (n == 0) return 1.0;
        
        double r = MyPow(x, n / 2);
        if (n % 2 == 0) return r * r;
        if (n > 0) return r * r * x;
        return r * r / x;
    }
}